crates/katana/node-bindings/src/lib.rs

```diff
@@ -176,20 +176,21 @@ pub struct Katana {
     json_log: bool,
     block_time: Option<u64>,
     db_dir: Option<PathBuf>,
-    rpc_url: Option<String>,
+    l1_provider: Option<String>,
     fork_block_number: Option<u64>,
     messaging: Option<PathBuf>,
 
     // Metrics options
-    metrics: Option<String>,
+    metrics_addr: Option<SocketAddr>,
+    metrics_port: Option<u16>,
 
     // Server options
-    port: Option<u16>,
-    host: Option<String>,
-    max_connections: Option<u64>,
-    allowed_origins: Option<String>,
+    http_addr: Option<SocketAddr>,
+    http_port: Option<u16>,
+    rpc_max_connections: Option<u64>,
+    http_cors_domain: Option<String>,
```

Comment:
```suggestion
    http_cors_domains: Option<String>,
```
can't the input contains several values separated by commas? If yes, wdyt about keeping it plural?

Reply:
I would go with `cors_origins` actually, wdyt?
<img width="805" alt="image" src="https://github.com/user-attachments/assets/bfc718df-ac45-4018-933e-32132e6bf879">

Reply:
yeah definitely remove the `http_` so that it's not confused with the http listen address/port

Reply:
At least on the CLI arg (ie `--http.corsdomain`), I prefixed it with `http` to indicate that the cors domains are only applied to the http server as we may introduce different transports (eg ws) and the idea is to allow configuring cors for each server.

Im a bit premature optimizing for this because its on the cli side. But in the code snippet above, i dont mind renaming it to what you guys suggested. Just want to make sure people are aware of my intention.